File: app/src/main/java/com/dominikgaller/alpinebooking/guide/inbound/rest/request/StartTourRequest.java
package com.dominikgaller.alpinebooking.guide.inbound.rest.request;

import java.time.Instant;

/**
 * REST request body for UC05 – StartTour.
 *
 * <p>{@code startedAt} is optional. If absent, the application service defaults to
 * the current time via {@code ClockPort}.
 *
 * <p>SDD: See {@code documentation/use-cases/uc05-start-tour.spec.md}, section 2.
 */
public record StartTourRequest(Instant startedAt) {
}

File: app/src/main/kotlin/com/example/service/guide/inbound/rest/request/StartTourRequest.kt
package com.example.service.guide.inbound.rest.request

import java.time.Instant

/**
 * REST request body for UC05 — StartTour. Optional.
 *
 * @property startedAt when the tour actually began. Null, or an absent body entirely, means
 *   "now" and the driver reads `ClockPort`.
 *
 * SDD: see `documentation/use-cases/uc05-start-tour.spec.md` § 9.
 */
data class StartTourRequest(
    val startedAt: Instant?,
)
